zgw/portaaltaak/src/main/kotlin/com/ritense/portaaltaak/PortaalTaakEventListener.kt

```diff
@@ -0,0 +1,178 @@
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.portaaltaak
+
+import com.fasterxml.jackson.core.JsonPointer
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.ritense.document.domain.impl.request.ModifyDocumentRequest
+import com.ritense.document.service.DocumentService
+import com.ritense.notificatiesapi.event.NotificatiesApiNotificationReceivedEvent
+import com.ritense.notificatiesapi.exception.NotificatiesException
+import com.ritense.objectenapi.ObjectenApiPlugin
+import com.ritense.objectmanagement.domain.ObjectManagement
+import com.ritense.objectmanagement.service.ObjectManagementService
+import com.ritense.plugin.domain.PluginConfigurationId
+import com.ritense.plugin.service.PluginService
+import com.ritense.processdocument.domain.impl.CamundaProcessInstanceId
+import com.ritense.processdocument.service.ProcessDocumentService
+import com.ritense.valtimo.service.CamundaProcessService
+import java.net.MalformedURLException
+import java.net.URI
+import org.camunda.bpm.engine.TaskService
+import org.camunda.bpm.engine.impl.persistence.entity.TaskEntity
+import org.camunda.bpm.engine.task.Task
+import org.springframework.context.event.EventListener
+import org.springframework.http.HttpStatus
+
+class PortaalTaakEventListener(
+    private val objectManagementService: ObjectManagementService,
+    private val pluginService: PluginService,
+    private val processDocumentService: ProcessDocumentService,
+    private val processService: CamundaProcessService,
+    private val taskService: TaskService,
+    private val documentService: DocumentService
+) {
+
+    @EventListener(NotificatiesApiNotificationReceivedEvent::class)
+    fun processCompletePortaalTaakEvent(event: NotificatiesApiNotificationReceivedEvent) {
+        val objectType = event.kenmerken["objectType"]
+
+        if (!event.kanaal.equals("objecten", ignoreCase = true) ||
+            !event.actie.equals("update", ignoreCase = true) ||
+            objectType == null
+        ) {
+            return
+        }
+
+        val objectManagement =
+            objectManagementService.findByObjectTypeId(objectType.substringAfterLast("/")) ?: return
+
+        // todo need clarification why this is really needed
+        pluginService.createInstance(PortaaltaakPlugin::class.java) { properties: JsonNode
+            ->
+            properties.get("objectManagementConfigurationId").textValue().equals(objectManagement.id.toString())
```

Comment:
Shouldn't this already be a UUID? If so, a regular equals should be sufficient.

Reply:
properties.get() returns a JsonNode. objectManagement.id is a UUID
not sure how equals works in kotlin